crates/oxc_parser/src/ts/statement.rs

```diff
@@ -510,4 +512,44 @@ impl<'a> Parser<'a> {
         let expr = self.with_context(Context::Decorator, Self::parse_lhs_expression)?;
         Ok(self.ast.decorator(self.end_span(span), expr))
     }
+
+    pub fn eat_modifiers_before_declaration(&mut self) -> Result<ModifierFlags> {
+        let mut flags = ModifierFlags::empty();
+        let mut modifiers = self.ast.new_vec();
+        while let Ok(modifier_flag) = self.cur_kind().try_into() {
+            let span = self.start_span();
+            // Const can only modify type parameters.
+            if modifier_flag == ModifierFlags::CONST {
+                break;
+            }
+            flags.set(modifier_flag, true);
+            let kind = self.cur_kind();
+            self.bump_any();
+            modifiers.push(Self::modifier(kind, self.end_span(span)));
+        }
```

Comment:
Currently we just eagerly eat all tokens that could be modifiers, ignoring the contextuality of some tokens. But to implement context check of modifiers is not trivial. Wondering if this is acceptable for now.